StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Elements/PaymentMethodElement/PaymentMethodElement.swift

```diff
@@ -16,15 +16,37 @@
  Other elements can rely on the default implementation provided in this file.
  */
 protocol PaymentMethodElement: Element {
-    /**
-     Modify the params according to your input, or return nil if invalid.
-     - Note: This is called on the Element hierarchy in depth-first search order.
-     */
+    /// Modify the params with default values.
+    ///
+    /// This method is called before `updateParams(params:)` and should be used to populate `params` with any necessary
+    /// default values, these can include values for fields not included in the element hierarchy.
+    func applyDefaults(params: IntentConfirmParams) -> IntentConfirmParams
+
+    /// Modify the params according to your input, or return nil if invalid.
+    /// - Note: This is called on the Element hierarchy in depth-first search order.
     func updateParams(params: IntentConfirmParams) -> IntentConfirmParams?
 }
 
 // MARK: - Default implementations
 extension ContainerElement {
+    func applyDefaults(params: IntentConfirmParams) -> IntentConfirmParams {
+        applyDefaultsHierarchichally(params: params)
```

Comment:
s/applyDefaultsHierarchichally/applyDefaultsHierarchically/ or maybe "recursively" is a more descriptive term?

Reply:
renamed